server/src/actions.js

```diff
@@ -0,0 +1,166 @@
+    }
+
+    applyPluginInstanceName(pluginInstanceId, pluginInstanceName) {
+        return this.database
+            .ref(`/v2/plugin_instances_named/${pluginInstanceId}`)
+            .set(pluginInstanceName)
+    }
+
+    applyInitialIndexToTopic(topicId) {
+        return this.database
+            .ref(`/v2/topics_index/${topicId}/current_index`)
+            .set(0)
+    }
+
+    applyHtmlToDeviceData(deviceId, html) {
+        return this.database
+            .ref(`/v2/devices_data/${deviceId}`)
+            .set(html)
+    }
+
+    applyHtmlToTopicPluginInstanceData(topicId, pluginInstanceId, html) {
+        return this.database
+            .ref(`/v2/topics_data/${topicId}/${pluginInstanceId}/html`)
+            .set(html)
+    }
+
+    applyPluginInstanceDataHtml(pluginInstanceId, html) {
+        return this.database
+            .ref(`/v2/plugin_instances_data/${pluginInstanceId}`)
+            .set(html)
+    }
+
+    queryPluginInstanceData(pluginInstanceId) {
+        return this.database
+            .ref(`/v2/plugin_instances_data/${pluginInstanceId}`)
+            .once('value')
+            .then(snapshot => snapshot.val())
+    }
+
+    queryPluginInstanceName(pluginInstanceId) {
+        return this.database
+            .ref(`/v2/plugin_instances_named/${pluginInstanceId}`)
+            .once('value')
+            .then(snapshot => snapshot.val())
+    }
+
+    queryTopicHtml(topicId, index) {
+        return this.database
+            .ref(`/v2/topics_data/${topicId}/`)
+            .once('value')
+            .then(snapshot => snapshot.val())
+            .then(instances => {
+                const instanceKey = Object.keys(instances)[index]
+                return instances[instanceKey].html
+            })
+    }
+
+    queryDeviceIdsByTopic(topicId) {
+        return this.database
+            .ref(`/v2/topic_to_devices/${topicId}`)
+            .once('value')
+            .then(readNonEmptySnapshot)
+            .then(Object.keys)
+    }
+
+    queryTopicCurrentIndex(topicId) {
+        return this.database
+            .ref(`/v2/topics_index/${topicId}/current_index`)
+            .once('value')
+            .then(snapshot => snapshot.val())
+    }
+
+    queryTopicIdsForPluginInstance(pluginInstanceId) {
+        return this.database
+            .ref(`/v2/plugin_instance_to_topic/${pluginInstanceId}`)
+            .once('value')
+            .then(readNonEmptySnapshot)
+            .then(Object.keys)
+    }
+
+    queryTopicsForAllDevices() {
+        return this.database
+            .ref(`/v2/topic_to_devices/`)
+            .once('value')
+            .then(snapshot => snapshot.val())
+    }
+
+    updatePlugin(plugin) {
+        return this.database
+            .ref('/v2/plugins')
+            .update(plugin)
+    }
+
+    removeTopic(topicId) {
+        const removeTopicIndex = this.database
+            .ref(`/v2/topics_index/${topicId}`)
+            .remove()
+            .catch(console.err)
```

Comment:
`console.error`? same for the others

Reply:
lol did this work? :sweat_smile:

Reply:
javascript :trollface:

Reply:
console.err definitely doesn't exist :joy: fixed https://github.com/novoda/dashboards/pull/30/commits/df4daea39eb9544056b0ff1761c8b325be659396